mongo/collection.go

```diff
@@ -508,6 +508,13 @@ func (coll *Collection) updateOrReplace(ctx context.Context, filter bsoncore.Doc
 	if uo.Upsert != nil {
 		updateDoc = bsoncore.AppendBooleanElement(updateDoc, "upsert", *uo.Upsert)
 	}
+	if uo.Hint != nil {
```

Comment:
Can we use the `createUpdateDoc` function here? Seems like a decent amount of code duplication.

Reply:
done